qiling/os/memory.py

```diff
@@ -302,16 +315,17 @@ def map(self, addr, size, perms=UC_PROT_ALL, ptr = None):
         '''
         if ptr == None:
             if self.is_mapped(addr, size) == False:
-               self.add_mapinfo(addr, addr + size, 'rw-', "[mapped]")
-               self.ql.uc.mem_map(addr, size)
+               self.ql.uc.mem_map(addr, size, perms)
+               self.add_mapinfo(addr, addr+size, perms, info if info else "[mapped]")
```

Comment:
just to make sure, so now all the mapinfo will be handle by mem.map and not syscall, we need to make sure all these are os free.

Reply:
all add_mapinfo has been deleted except one in macos

Reply:
just removed it since we don't really use it right now

Reply:
maybe can take a quick look at macos ?

Reply:
i did , check this commit b86a5f6.
we are not using it right now, and its probably not goona work because there is no prot_mapping for macos, so I removed it.

Reply:
ok, let me check one more round later and commit